src/lib.rs

```diff
@@ -36,13 +36,36 @@
 
 macro_rules! resultify {
     ($ret:expr) => {
-        match $ret >= 0 {
-            true    => Ok($ret as _),
-            false   => Err(io::Error::from_raw_os_error(-$ret)),
+        match $ret {
```

Comment:
Could you instead just add an assignment statement (`let ret = $ret;`) so that this is more resilient to any future changes?

Reply:
Will do.